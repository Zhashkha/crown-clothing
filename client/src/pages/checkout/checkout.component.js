import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  CheckoutPageContainer,
  CheckoutHeaderContainer,
  HeaderBlockContainer,
  TotalContainer,
  WarningContaier,
} from "./checkout.styles";
import {
  selectCartItems,
  selectCartTotal,
} from "../../redux/cart/cart.selectors";
import CheckoutItem from "../../components/checkout-item/checkout-item.component";
import StripeCheckoutButton from "../../components/stripe-button/stripe-button.component";

const CheckoutPage = ({ cartItems, total }) => (
  <CheckoutPageContainer>
    <CheckoutHeaderContainer>
      <HeaderBlockContainer>Product</HeaderBlockContainer>
      <HeaderBlockContainer>Description</HeaderBlockContainer>
      <HeaderBlockContainer>Quantity</HeaderBlockContainer>
      <HeaderBlockContainer>Price</HeaderBlockContainer>
      <HeaderBlockContainer>Remove</HeaderBlockContainer>
    </CheckoutHeaderContainer>
    {cartItems.map((cartItem) => (
      <CheckoutItem key={cartItem.id} cartItem={cartItem} />
    ))}
    <TotalContainer>
      <span>TOTAL: ${total}</span>
    </TotalContainer>
    <WarningContaier>
      *Please use the follwing test credit card for payments*
      <br />
      4242 4242 4242 4242 - Exp: 12/20 - CVC: 123
    </WarningContaier>
    <StripeCheckoutButton price={total} />
  </CheckoutPageContainer>
);

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal,
});

export default connect(mapStateToProps)(CheckoutPage);
