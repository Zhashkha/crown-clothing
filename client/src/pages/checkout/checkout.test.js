import React from "react";
import { shallow } from "enzyme";

import { CheckoutPage } from "./checkout.component";

let wrapper;
beforeEach(() => {
  const mockProps = { cartItems: [], total: 100 };

  wrapper = shallow(<CheckoutPage {...mockProps} />);
});

it("test CheckoutPage snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});
