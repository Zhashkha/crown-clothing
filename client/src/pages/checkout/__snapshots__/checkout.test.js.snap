// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CheckoutPage snapshot 1`] = `ShallowWrapper {}`;
